File: src/015/p015.swift
func combination(_ n: Int, _ k: Int) -> Int {
    return (0...k-1).reduce(1) { (acc: Int, i: Int) in acc * (n - i) / (i + 1) }
}

func solve() -> Int {
    return combination(40, 20)
}

let result = solve()
print(result)

File: src/015/p015.clj
(def init [1 1])

(defn step [line] 
  (->> (concat [0] line [0])
       (partition 2 1)
       (mapv (partial apply +))))

(defn solve [] 
  (->> init
      (iterate step)
      (drop 19)
      first
      (reduce ((map #(* % %)) +))))

(prn (solve))
